controllers/registeredservice_controller.go

```diff
@@ -56,7 +65,36 @@ func (r *RegisteredServiceReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	log := log.FromContext(ctx)
 
 	var rs primazaiov1alpha1.RegisteredService
-	if err := r.Client.Get(ctx, req.NamespacedName, &rs); err != nil {
+	err := r.Client.Get(ctx, req.NamespacedName, &rs)
+	if err != nil && errors.IsNotFound(err) {
+		log.Info("Registered Service not found, handling delete event")
```

Comment:
I'm not convinced this is right; AIUI if you want deletion events to show up in your controller, you need to set a finalizer.  See [here](https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/) for more details.

Reply:
This seems to work since i don't need data from the object, obviously I cannot get it cause that is how I get in the error state, but if you notice the req.name has the right name for the request. I was thinking this is an easier way to implement initially. If for some reason I need data in the resource being delete I have not choice but to implement a finilizer